examples/demo-app/src/actions.js

```diff
@@ -365,27 +416,121 @@ export function exportFileToCloud(providerName) {
+      type: 'application/json',
+      blob: file,
+      name: extraData.title,
+      description: extraData.description,
+      isPublic,
+      cloudProvider
+    })
+    // need to perform share as well
+    .then(
+      response => {
+        if (cloudProvider.shareFile) {
+          const responseUrl = cloudProvider.getMapPermalink
+            ? cloudProvider.getMapPermalink(response.url, false)
+            : getMapPermalink(response.url, false)
+          dispatch(push(responseUrl));
+        }
+        dispatch(setPushingFile(false, {
+          filename: file.name,
+          status: 'success',
+          metadata: response,
+          provider: cloudProvider.name
+        }));
+        dispatch(toggleModal(null));
+      },
+      error => {
+        dispatch(setPushingFile(false, {
+          filename: file.name,
+          status: 'error',
+          error, provider:
+          cloudProvider.name
+        }));
+      }
+    );
+  };
+}
+
+/**
+ * This method will export the current kepler config file to the choosen cloud backend platform
+ * @param providerName
+ * @param data
+ * @returns {Function}
+ */
+export function saveMapToCloud(providerName, {map, info, thumbnail}) {
+  return (dispatch) => {
+    if (!providerName) {
+      dispatch(showNotification(DEFAULT_NOTIFICATION_TYPES.error, 'No cloud provider selected.'));
+
+      throw new Error('No cloud provider selected.')
+    }
+
+    const cloudProvider = getCloudProvider(providerName);
+    const data = JSON.stringify(map);
+    const newBlob = new Blob([data], {type: 'application/json'});
+    const fileName = `/keplergl_${generateHashId(6)}.json`;
+    info.title = info.title || fileName;
     const file = new File([newBlob], fileName);
     // We are gonna pass the correct auth token to init the cloud provider
-    dispatch(setPushingFile(true, {filename: file.name, status: 'uploading', metadata: null}));
+    dispatch(setPushingFile(true, {
+      filename: file.name,
+      status: 'uploading',
+      metadata: null,
+      provider: cloudProvider.name
+    }));
     cloudProvider.uploadFile({
       data,
       type: 'application/json',
       blob: file,
-      name: fileName,
-      isPublic: true,
+      name: info.title,
+      description: info.description,
+      thumbnail,
+      isPublic: false,
       cloudProvider
     })
     // need to perform share as well
     .then(
       response => {
         if (cloudProvider.shareFile) {
-          dispatch(push(`/${MAP_URI}${response.url}`));
+          const responseUrl = cloudProvider.getMapPermalink
+            ? cloudProvider.getMapPermalink(response.url, false)
+            : getMapPermalink(response.url, false)
+          dispatch(push(responseUrl));
         }
-        dispatch(setPushingFile(false, {filename: file.name, status: 'success', metadata: response}));
+        dispatch(setPushingFile(false, {
+          filename: file.name,
+          status: 'success',
+          metadata: response,
```

Comment:
Is it necessary for kepler to store the entire response here? looks like kepler only needs the `shareUrl` and `folderLink` to display. And it sends the url back to `provider.getMapPermalink`. Why don't the provider store `shareUrl` and `folderLink` when request succeed, and kepler only call `provider.getMapPermalink()` to render it. 

If this makes sense. I will implement it in my pr

Reply:
This reponse object only has a `url` field from CARTO provider (same as `shareUrl`) but I kept the entire response in case of other providers need it.
Anyway, I agree with your proposal, we already save last saved/loaded visualization and offer a method called `getCurrentVisualization` to get it, but also offer a `getMapPermalinkFromParams` method to be able to create a permalink to any vis from the load map modal or from query params.